third_party/terraform/data_sources/data_source_google_compute_backend_bucket.go

```diff
@@ -0,0 +1,37 @@
+package google
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func dataSourceGoogleComputeBackendBucket() *schema.Resource {
+	dsSchema := datasourceSchemaFromResourceSchema(resourceComputeBackendBucket().Schema)
+
+	// Set 'Required' schema elements
+	addRequiredFieldsToSchema(dsSchema, "name")
+
+	// Set 'Optional' schema elements
+	addOptionalFieldsToSchema(dsSchema, "project")
+
+	return &schema.Resource{
+		Read:   dataSourceComputeBackendBucketRead,
+		Schema: dsSchema,
+	}
+}
+
+func dataSourceComputeBackendBucketRead(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	serviceName := d.Get("name").(string)
```

Comment:
nit - maybe we should choose something otehr than service name here/in test? i'm fine if you want to keep it as is/choose something else since bucketName also sounds like the GCS bucket field, but serviceName sounds weird if you haven't done LB stuff before? maybe backendName